release-content/migration-guides/generic-option-parameter.md

```diff
@@ -0,0 +1,28 @@
+---
+title: Generic `Option` Parameter
+pull_requests: [18766]
+---
+
+`Option<Single<D, F>>` will now resolve to `None` if there are multiple entities matching the query.
+Previously, it would only resolve to `None` if there were no entities, and would skip the system if there were multiple.
+
+We have introduced a blanket `impl SystemParam for Option` that resolves to `None` if the parameter is invalid.
+This allows third-party system parameters to work with `Option`, and makes the behavior more consistent.
+
+If you want a system to run when there are no matching entities but skip when there are multiple,
+you will need to use `Query<D, F>` and call `single()` yourself.
+
+```rust
+// 0.16
+fn my_system(single: Option<Single<&Player>>) {
```

Comment:
Couldn't the 0.17 version of this use a result instead of an option and then just check it directly without needing to do the more-line query form?

Reply:
Not easily, no.  The only way to use `Result` as a system parameter is `Result<T, SystemParamValidationError>`.  `SystemParamValidationError` gives you strings for an error message and a flag distinguishing skipping and failing parameters, but no way to recover a `QuerySingleError`.  (The `Result` impl is mainly intended for generic systems where you want to bubble up parameter validation conditionally.)  

Were you relying on the old behavior of distinguishing the two kinds of errors?